### ISSUE TITLE
Add list of preserved objects

### ISSUE THREAD

**Issue by bondjimbond** [bondjimbond/islandora_westvault#43]
Currently we have a link to a "Preserved objects list" page (admin/islandora/tools/westvault/objects), but it doesn't do anything.

It should produce the following:
- Minimum: A table of objects with a PRESERVATION datastream
- Ideal: Include in that table objects that have already been passed to Westvault (found by reading the PRESERVATION datastream).

**Comment by mjordan** [comment]
Any reason this couldn't be done with Views? I'm just thinking about when you have 1000+ items...

**Comment by bondjimbond** [comment]
Two reasons:

1. I don't know how to do it (insert a View using code)
2. I'm not sure that the XML-reading function at https://github.com/bondjimbond/islandora_westvault/pull/44/files#diff-7b756ae198963ede91c86d52f1385db8R147 to produce the "PRESERVED" or "PENDING" result could be done with Views

Your thoughts?

**Comment by mjordan** [comment]
I wouldn't be too worried about the first question. All I'm saying is that if you just produce a table, you're eventually going to need to deal with pagination, etc. which might be a hassle.

By default, all .xml datastreams are slurped into their own Solr field, but that won't be of much use, unless you can do a string query for "PRESERVED" in the field the PRESERVATION datastream is indexed into (which I think is in fact possible in Views). Ideally, you would index all the elements in the PRESERVATION ds, but to get that to work, you need to add an XSLT to your gsearch configs, etc. so that's probably a no go.

So, maybe using the current approach is OK for now. But I stand by my warning about the future.

**Comment by bondjimbond** [comment]
Yeah, if I could do this in Views I'd be a lot happier -- could add exposed filters and sort, for one. But while it's pretty easy to create a View that lists all objects with a PRESERVATION datastream... the question of how to read that datastream remains.

**Comment by mjordan** [comment]
I don't know without getting my hands dirty. Maybe a query to the group will yield some suggestsions?

**Comment by mjordan** [comment]
Just checked on my vagrant and the content of the PRESERVATION ds is not being slurped into a solr field like I thought. I was sure that all XML datastream content was indexed as a blob at the time I was developing https://github.com/SFULibrary/islandora_solution_pack_xml. Maybe the default gsearch configs have changed since then.

**Comment by bondjimbond** [comment]
@mjordan Yeah, it only slurps the metadata about the PRESERVATION datastream, not the content. I think you would have to write in custom datastreams specially to get their content slurped up.

**Comment by bondjimbond** [comment]
Thought: would it be possible to do this with Views PHP?

- Get a list of objects with a PRESERVATION datastream
- Use a Views PHP field to read the datastream content
- Print the PRESERVED/PENDING etc. status in a table column
- Filter on that custom field

I'll have to do some experiments to see if this is plausible... If so, I may have questions about how to integrate a programmed View into a module.
